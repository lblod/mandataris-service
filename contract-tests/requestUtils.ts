export let sessionUri = '';

export async function setSession(newSessionUri: string) {
  sessionUri = newSessionUri;
}

export async function userRequest(
  method: string,
  url: string,
  body: unknown = undefined,
  extraHeaders = {},
) {
  const response = await fetch(url, {
    method,
    headers: {
      'Content-Type': 'application/json',
      'mu-session-id': sessionUri,
      ...extraHeaders,
    },
    body: body ? JSON.stringify(body) : undefined,
  });
  const responseBody = await response.text();
  const headers = response.headers;
  const status = response.status;
  if (process.env.LOG_RESPONSES) {
    console.log(`Response from ${url}:
      status: ${status}
      body: ${body}`);
  }
  return {
    status,
    body: responseBody,
    headers,
  };
}
