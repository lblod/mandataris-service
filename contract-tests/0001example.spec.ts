import 'dotenv/config';

import { describe, expect, test, beforeAll } from '@jest/globals';
import { setSession, userRequest } from './requestUtils';

describe('index', () => {
  beforeAll(async () => {
    setSession(
      'http://mu.semte.ch/sessions/a3bacf3e-fb4c-11ef-851c-0242ac120016',
    );
  }, 25000);
  test('hello world', async () => {
    const { body } = await userRequest(
      'GET',
      'http://target/mandatarissen/6729D1A2A12BA678FC88719E/fracties',
    );
    console.log('Deltas', globalThis.deltas);
    expect(body).toMatchSnapshot();
  });

  test('hello world2', async () => {
    expect(1 + 2).toBe(2);
  });
});
