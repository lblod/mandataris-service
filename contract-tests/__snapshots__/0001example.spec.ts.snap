// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index hello world 1`] = `"{"fracties":["6729D13BA12BA678FC887193"]}"`;
